spec/abstractops.html

```diff
@@ -1800,6 +1800,27 @@ <h1>
     </emu-alg>
   </emu-clause>
 
+  <emu-clause id="sec-tostringifstringorobject" type="abstract operation">
+    <h1>
+      ToStringIfStringOrObject (
+        _argument_: an ECMAScript language value,
+      ): either a normal completion containing a String, or a throw completion
+    </h1>
+    <dl class="header">
+      <dt>description</dt>
+      <dd>
+        It returns a String, either by returning _argument_ if it is already a String, or by converting _argument_ to a String if it is an Object.
+        However, if _argument_ is a non-String primitive value, then it will throw a TypeError.
+        This abstract operation is used when coercing Numbers or other primitive values to Strings is undesirable.
+      </dd>
+    </dl>
+    <emu-alg>
+      1. If Type(_argument_) is Object, return ? ToString(_argument_).
```

Comment:
```suggestion
      1. If Type(_argument_) is Object, set _argument_ to ? ToPrimitive(_argument_, ~string~).
```
and i'd perhaps rename the AO to `ToStringOrThrow`?

Reply:
Any of these approaches provide the semantics I prefer, of course! The advantage of the AO is indeed discoverability.